packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -126,6 +139,8 @@ export abstract class SfdxCommandletExecutor<T>
   }
 
   public abstract build(data: T): Command;
+  protected getRemoteChanges?(): StatusOutputRowType[] | undefined;
+  protected updateCacheAfterPushPull?(): void;
```

Comment:
nitty suggestion, you could have this method be called updateCache and just have it default to a noop function, then override it in the push/pull case.  Then you wouldn't have the sometimes defined method you'd just alway execute it.